xdebug_code_coverage.c

```diff
@@ -1056,38 +1056,18 @@ static int prefill_from_function_table(zend_op_array *opa)
 # define XDEBUG_PTR_KEY_FMT "%016lX"
 #endif
 
-
-static int mark_class_as_visited(zend_class_entry *ce)
-{
-	int     already_visited = 0;
-	void   *dummy; /* we only care about key existence, not value */
-	char    key[XDEBUG_PTR_KEY_LEN + 1];
-
-	snprintf(key, XDEBUG_PTR_KEY_LEN + 1, XDEBUG_PTR_KEY_FMT, (uintptr_t) ce);
-
-	if (xdebug_hash_find(XG(visited_classes), key, XDEBUG_PTR_KEY_LEN, (void*) &dummy)) {
-		already_visited = 1;
-	} else {
-		xdebug_hash_add(XG(visited_classes), key, XDEBUG_PTR_KEY_LEN, NULL);
-	}
-
-	return already_visited;
-}
-
 static int prefill_from_class_table(zend_class_entry *ce)
 {
 	if (ce->type == ZEND_USER_CLASS) {
-		if (!mark_class_as_visited(ce)) {
-			zend_op_array *val;
+		zend_op_array *val;
 
-			xdebug_zend_hash_apply_protection_begin(&ce->function_table);
+		xdebug_zend_hash_apply_protection_begin(&ce->function_table);
```

Comment:
Side note: What's the apply protection here for?

Reply:
I don't really know to be honest. AFAIK this is only for recursion protection, right?

Reply:
That's right. There should be not possibility for recursion here.